src/crafting.h

```diff
@@ -69,6 +69,8 @@ struct recipe {
         bool hidden;
     };
     std::vector<bookdata_t> booksets;
+    std::set<std::string> flags;
```

Comment:
The reason to use a bitset instead of a set of strings is if the code doing the lookup is performance critical, but this requires the code to maintain a mapping of strings to numeric values, so it should only be done if necessary.